### ISSUE TITLE
XLights 2023.20

### ISSUE THREAD

**Issue by Rlbos** [smeighan/xLights#4184]
Upgraded Xlights 2023.20 every time I open the save light is on. I haven't change anything on the desktop I have to render my sequences.

[Document 4.pdf](https://github.com/smeighan/xLights/files/13537069/Document.4.pdf)

**Comment by dartunder** [comment]
This happens sometimes for some configurations.  For example, if universe per string is enabled.

**Comment by dkulp** [comment]
Double check with .21.  I fixed at least one of these with .21, but that was related to using a base folder.

**Comment by dartunder** [comment]
I don't know about his case, but the code forces a recalculation that triggers the save button if the universe per string is enabled. His situation may be different.

My case is being triggered ( outputManager->SomethingChanged()) by this code from ModeManager.cpp line 939

 if (it->IsAutoSize()) {
            auto eth = dynamic_cast<const ControllerEthernet*>(it);
            if (it->GetChannels() != std::max((int32_t)1, (int32_t)ch - 1) || (eth != nullptr && eth->IsUniversePerString())) {
                logger_zcpp.debug("    Resizing output to %d channels.", std::max((int32_t)1, (int32_t)ch - 1));

                auto oldC = it->GetChannels();
                // Set channel size won't always change the number of channels for some protocols
                it->SetChannelSize(std::max((int32_t)1, (int32_t)ch - 1), allSortedModels);
                if (it->GetChannels() != oldC || (eth != nullptr && eth->IsUniversePerString())) {
                    outputManager->SomethingChanged();

                    xlights->GetOutputModelManager()->AddASAPWork(OutputModelManager::WORK_NETWORK_CHANGE, "ReworkStartChannel");
                    xlights->GetOutputModelManager()->AddASAPWork(OutputModelManager::WORK_NETWORK_CHANNELSCHANGE, "ReworkStartChannel");
                    xlights->GetOutputModelManager()->AddASAPWork(OutputModelManager::WORK_UPDATE_NETWORK_LIST, "ReworkStartChannel");
                    xlights->GetOutputModelManager()->AddASAPWork(OutputModelManager::WORK_SAVE_NETWORKS, "ReworkStartChannel");
                    xlights->GetOutputModelManager()->AddASAPWork(OutputModelManager::WORK_RESEND_CONTROLLER_CONFIG, "ReworkStartChannel");
                }
            }
        }

**Comment by derwin12** [comment]
He is/was in zoom..  happens in .21 nightly.
[xLightsProblem.zip](https://github.com/smeighan/xLights/files/13537169/xLightsProblem.zip)

**Comment by dartunder** [comment]
Yes, he has a universe per string enabled on his hinks pix, so that is the same issue.   
UPS="TRUE"

**Comment by derwin12** [comment]
Confirmed .. removing Universe per string stops it being flagged to save.

**Comment by Rlbos** [comment]
unchecking Universe per string fixed that issue.  Will test HinkPix controller.

**Comment by Rlbos** [comment]
I updated to .21 still the same proble
On Sat, Dec 2, 2023 at 13:11, Daniel Kulp ***@***.***> wrote:
	Double check with .21. I fixed at least one of these with .21, but that was related to using a base folder.

**Comment by cybercop23** [comment]
@Rlbos originally you said it worked when you unchcked UPS. Then tested 2023.21 and "same proble". What's that exactly? Is UPS still checked? If you uncheck it and it doesn't force a re-render then it seems like that's what you want/need. 
Also, plese try the latest 2024.2.
If working as explained above, please close.
